pages/ChatGpt.jsx

```diff
@@ -0,0 +1,89 @@
+import { useState } from 'react'
+import '@chatscope/chat-ui-kit-styles/dist/default/styles.min.css';
+import { MainContainer, ChatContainer, MessageList, Message, MessageInput, TypingIndicator } from '@chatscope/chat-ui-kit-react';
+
+// Upload your chatgpt api key here
+const API_KEY = process.env.NEXT_PUBLIC_API_KEY;
+
+const systemMessage = {
+  "role": "system", "content": "Explain things like you're talking to a software professional with 2 years of experience."
+}
+
+const ChatGpt = ({ data }) => {
+  const [messages, setMessages] = useState([
+    {
+      message: "Hello, I'm ChatGPT! Ask me anything!",
+      sentTime: "just now",
+      sender: "ChatGPT"
+    }
+  ]);
+  const [isTyping, setIsTyping] = useState(false);
+
+  const handleSend = async (message) => {
+    const newMessage = {
+      message,
+      direction: 'outgoing',
+      sender: "user"
+    };
+
+    const newMessages = [...messages, newMessage];
+    setMessages(newMessages);
+    setIsTyping(true);
+
+    await processMessageToChatGPT(newMessages, data);
+  };
+
+  async function processMessageToChatGPT(chatMessages, data) {
+    let apiMessages = chatMessages.map((messageObject) => {
+      let role = messageObject.sender === "ChatGPT" ? "assistant" : "user";
+      return { role: role, content: messageObject.message}
+    });
+
+    let dataContent = typeof data === "string" ? data : JSON.stringify(data);
+
+
+    const apiRequestBody = {
+      "model": "gpt-3.5-turbo",
+      "messages": [
+        systemMessage,
+        { role: "system", content: dataContent }, // passing data
+        ...apiMessages 
+      ]
+    }
+
+    await fetch("https://api.openai.com/v1/chat/completions", 
+    {
+      method: "POST",
+      headers: {
+        "Authorization": "Bearer " + API_KEY,
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(apiRequestBody)
+    }).then(data => data.json()).then(data => {
+      console.log(data); 
```

Comment:
This is good for demo and debugging, however this console.log will be in production delivery, so you can try to use redux-logger: https://github.com/LogRocket/redux-logger , or just simply use sth. like `process.env.NODE_ENV` e.g.
```js
if (process.env.NODE_ENV === 'development') {
  console.log(...);
}
```

T